nexus/db-queries/src/db/datastore/affinity.rs

```diff
@@ -703,6 +842,129 @@ impl DataStore {
         Ok(())
     }
 
+    async fn anti_affinity_group_member_add_group(
+        &self,
+        opctx: &OpContext,
+        authz_anti_affinity_group: &authz::AntiAffinityGroup,
+        affinity_group_id: AffinityGroupUuid,
+    ) -> Result<(), Error> {
+        let err = OptionalError::new();
+        let conn = self.pool_connection_authorized(opctx).await?;
+        self.transaction_retry_wrapper("anti_affinity_group_member_add_group")
+            .transaction(&conn, |conn| {
+                let err = err.clone();
+                use db::schema::anti_affinity_group::dsl as anti_affinity_group_dsl;
+                use db::schema::affinity_group::dsl as affinity_group_dsl;
+                use db::schema::affinity_group_instance_membership::dsl as a_instance_membership_dsl;
+                use db::schema::anti_affinity_group_affinity_membership::dsl as aa_affinity_membership_dsl;
+                use db::schema::sled_resource_vmm::dsl as resource_dsl;
```

Comment:
turbo nit, sorry: i usually expect all value-level code in a block to be below all the imports in that block, so i found the `let err = err.clone();` above the imports a bit easy to miss. i might prefer:
```suggestion
                use db::schema::anti_affinity_group::dsl as anti_affinity_group_dsl;
                use db::schema::affinity_group::dsl as affinity_group_dsl;
                use db::schema::affinity_group_instance_membership::dsl as a_instance_membership_dsl;
                use db::schema::anti_affinity_group_affinity_membership::dsl as aa_affinity_membership_dsl;
                use db::schema::sled_resource_vmm::dsl as resource_dsl;

                let err = err.clone();
```
on the other hand, i will be the first to admit that i'm probably being obnoxious about this, so feel free to disregard me.